### PR TITLE
Clarify that the event field of the send_join is only required when performing a restricted join

### LINKED ISSUE #1708: Clarify that join events only need to be signed by resident servers if using `join_authorised_via_users_server`
**Link to problem area**: https://spec.matrix.org/v1.9/server-server-api/#put_matrixfederationv2send_joinroomideventid

The join event only needs a signature from a resident server if joining via the restricted join rules flow, i.e. `join_authorised_via_users_server` is specified. 

c.f. [MSC3083](https://github.com/matrix-org/matrix-spec-proposals/blob/main/proposals/3083-restricted-rooms.md) and [Room Version 8 auth rules](https://spec.matrix.org/v1.9/rooms/v8/)

### PR DESCRIPTION
**Pull request by Kladki**
Fixes #1708 

### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [x] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview: https://pr1834--matrix-spec-previews.netlify.app

### CODE REVIEW

**Review by anoadragon453 — APPROVED** [review]
This looks correct to me given the discussion in https://github.com/matrix-org/matrix-spec/issues/1708.

Thanks for bringing this one over the line!

**Comment by clokep** [comment]
> This looks correct to me given the discussion in #1708.

I'm unsure it is. There's two definitions at play:

1. The issue says that the event *signature* is required only for a restricted join.
2. The change proposed says that the *event* itself is required only for a restricted join.

I think the intention was 1 to keep the API shape easier for implementations.

File: data/api/server-server/joins-v2.yaml [hunk 1]
```diff
@@ -207,9 +207,9 @@ paths:
                     title: SignedMembershipEvent
                     x-addedInMatrixVersion: "1.2"
                     description: |-
-                      Required if the room version [supports restricted join rules](/rooms/#feature-matrix). The signed
-                      copy of the membership event sent to other servers by the resident server, including the resident
-                      server's signature.
+                      Required if the `content` of the event in the request contained the `join_authorised_via_users_server`
+                        field. The signed copy of the membership event sent to other servers by the resident server,
```
Comment (richvdh) [inline data/api/server-server/joins-v2.yaml:211]: As I've just written over at #1708: this doesn't seem to match Synapse's behaviour. AFAICT, the presence of `join_authorised_via_users_server` is not sufficient for Synapse to return a (signed) event here.

Comment (richvdh) [reply data/api/server-server/joins-v2.yaml:211]: The synapse code is at https://github.com/element-hq/synapse/blob/v1.108.0/synapse/federation/federation_server.py#L952-L956. 

Currently, it returns the event for all requests to `/send_join`. But it only *signs* the event if the user is actually performing a restricted join.

**Comment by richvdh** [comment]
This PR is updated by #1840

**Pull request merged**